src/PooledArrays.jl

```diff
@@ -201,6 +208,16 @@ function Base.copy!(dest::PooledArray{T, R, N, RA},
     return dest
 end
 
+function Base.copy!(dest::PooledArray{T, R, N, RA},
+                    src::SubArray{T, N, PooledArray{T, R, M, RA}}) where {T, R, N, M, RA}
+    copy!(dest.refs, view(parent(src).refs, src.indices))
```

Comment:
A trick that I used in CategoricalArrays to avoid duplicating methods is to define `refs(x::SubArray{...}) = view(parent(x).refs, x.indices)` and `refs(x::PooledArray) = x.refs`. Then the same method can work both both. Same for `copyto!`.

Reply:
I added DataAPI.jl support for `SubArray{<:Any, <:Any, <:PooledArray}`.